src/modules/decklink/producer/decklink_producer.cpp

```diff
@@ -345,7 +341,7 @@ class decklink_producer : public IDeckLinkInputCallback
         graph_->set_text(print());
         diagnostics::register_graph(graph_);
 
-        if (FAILED(input_->EnableVideoInput(mode_->GetDisplayMode(), bmdFormat8BitYUV, 0))) {
+        if (FAILED(input_->EnableVideoInput(mode_->GetDisplayMode(), bmdFormat8BitYUV, bmdVideoInputEnableFormatDetection))) {
```

Comment:
I wouldn't be surprised if this call fails when the card doesn't support format detection, so it would be nice to check that.

Reply:
I was curious and have some cards I suspected do not support format detection and can confirm that this fails with that card.

```
[2018-09-08 19:25:15.785] [info]    Received message from Console: play 1-10 decklink 1\r\n
[2018-09-08 19:25:15.831] [warning] Device does not support video-format: 1080i50
[2018-09-08 19:25:15.860] [warning] Decklink producer not supporting detection
[2018-09-08 19:25:15.871] [error]   Exception: c:\code\casparcg22\src\modules\decklink\producer\decklink_producer.cpp(351): Throw in function __cdecl caspar::decklink::decklink_producer::decklink_producer(const struct caspar::core::video_format_desc &,int,const class caspar::spl::shared_ptr<class caspar::core::frame_factory> &,const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,const class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,bool)
[2018-09-08 19:25:15.871] [error]   Dynamic exception type: class boost::exception_detail::clone_impl<struct caspar::caspar_exception>
[2018-09-08 19:25:15.871] [error]   [struct boost::errinfo_api_function_ * __ptr64] = EnableVideoInput
[2018-09-08 19:25:15.871] [error]   [struct caspar::tag_msg_info * __ptr64] = Intensity Pro [1|1080i5000] Could not enable video input.
```

It's easy to check for suppport though, I used the following to add a temporary line to the log
```
int status = 0;
        if (FAILED(attributes_->GetFlag(BMDDeckLinkSupportsInputFormatDetection, &status)) || !status) {
            CASPAR_LOG(warning) << L"Decklink producer not supporting detection";
        }
```

Reply:
Hey @Julusian, thanks for testing! I'll add that to the PR when checked the other input formats / conversions.